### PR TITLE
Sample format checks

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #5101 
Resolves: #4752

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
macOS build is failing

**Comment by vsverchinsky** [comment]
> macOS build is failing

Now fixed

File: src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp [hunk 1]
```diff
@@ -762,6 +762,16 @@ void WaveTrackMenuTable::OnMergeStereo(wxCommandEvent &)
            .IconStyle(Icon::Error));
       return;
    }
+   if(pTrack->GetSampleFormat() != partner->GetSampleFormat())
+   {
+      BasicUI::ShowMessageBox(XO(
+"Mono tracks must have the same sample format in order to be combined"
+"into a stereo track"),
```
Comment (Paul-Licameli) [inline src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp:769]: @LWinterberg

File: libraries/lib-wave-track/WaveTrack.h [hunk 2]
```diff
@@ -830,8 +830,8 @@ class WAVE_TRACK_API WaveTrack final
     */
    WaveClipHolders mClips;
 
-   sampleFormat  mFormat;
-   mutable int   mLegacyRate{ 0 }; //!< used only during deserialization
+   mutable int  mLegacyRate{ 0 }; //!< used only during deserialization
+   sampleFormat mLegacyFormat;
```
Comment (Paul-Licameli) [inline libraries/lib-wave-track/WaveTrack.h:834]: Please duplicate the comment

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 3]
```diff
@@ -114,6 +117,7 @@ struct WaveTrackData : ClientData::Cloneable<> {
    std::atomic<float> mPan{ 0.0f };
 
    int mRate{ 44100 };
+   sampleFormat mFormat;
```
Comment (Paul-Licameli) [inline libraries/lib-wave-track/WaveTrack.cpp:120]: Uninitialized in the default constructor?  Make it `floatSample`

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 4]
```diff
@@ -2152,6 +2183,12 @@ std::optional<TranslatableString> WaveTrack::GetErrorOpening() const
       return XO(
 "This project cannot be opened because it contains a stereo track with "
 "different sample rates. This functionality is no longer supported since the "
+"release of Audacity version 3.4");
+
+   if(!FormatConsistencyCheck())
+      return XO(
+"This project cannot be opened because it contains a stereo track with "
+"different sample formats. This functionality is no longer supported since the "
 "release of Audacity version 3.4");
```
Comment (Paul-Licameli) [inline libraries/lib-wave-track/WaveTrack.cpp:2192]: @LWinterberg wording?

**Comment by Paul-Licameli** [comment]
Done reviewing.  Just fix one initialization.

**Review by Paul-Licameli — APPROVED** [review]

**Comment by vsverchinsky** [comment]
Rebased onto recent master that has fix for misalligned channel, to avoid bugs that are already fixed from being observed

File: libraries/lib-wave-track/WaveTrack.cpp (hunk 4 above)
Comment (LWinterberg) [reply libraries/lib-wave-track/WaveTrack.cpp:2192]: looks good to me

File: src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp (hunk 1 above)
Comment (LWinterberg) [reply src/tracks/playabletrack/wavetrack/ui/WaveTrackControls.cpp:769]: looks good to me

**Comment by Paul-Licameli** [comment]
This is approved for merge.  I rebased it onto recent master and made the needed conflict resolutions.

**Pull request merged**